### PR TITLE
Simple mob sounds

### PR DESCRIPTION
**Pull request by Goldsteel**
This change permits simple mobs to audibly vocalise based on their emotes and speech. I've added sounds for cats and dogs but it is fairly easy to add more. There is a variable that just includes a list of paths to the sounds and making changes to that should work for everything.

:cl:
 * rscadd: Simple mobs now make sounds when talking and being touched.

### CODE REVIEW

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 1]
```diff
@@ -12,7 +12,8 @@
 	speak_emote = list("purrs", "meows")
 	emote_hear = list("meows", "mews")
 	emote_see = list("shakes its head", "shivers")
-	speak_chance = 1
+	emote_sound = list("sound/voice/catmeow.ogg","sound/voice/catpurr.ogg")
+	speak_chance = 100
```
Comment (MadmanMartian) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: !

Comment (Goldsteel) [reply code/modules/mob/living/simple_animal/friendly/cat.dm]: Oh god oh fuck

**Comment by DamianX** [comment]
Could use some sort of cooldown because fun is not to be tolerated

**Comment by Goldsteel** [comment]
I can pop in a sanity check, sure.

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 2]
```diff
@@ -24,6 +24,7 @@ var/global/list/animal_count = list() //Stores types, and amount of animals of t
 	var/speak_chance = 0
 	var/list/emote_hear = list()	//Hearable emotes
 	var/list/emote_see = list()		//Unlike speak_emote, the list of things in this variable only show by themselves with no spoken text. IE: Ian barks, Ian yaps
+	var/list/emote_sound = list()   //Plays a random sound when emote_hear activates
```
Comment (DamianX) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: This is not super-important but this comment is misleading: sounds in `emote_sound` also play when `speak` is "activated"

**Comment by DamianX** [comment]
Ackshually, after reading the code and not trusting your changelog, you say:
>Simple mobs now make sounds when talking

But the sound only has a chance to play when the mob randomly emotes, or when they're `attack_hand`'d, which should already be setting an attack delay.

It doesn't look like these changes make the mob play a sound when a player controlling it says something, which is what the changelog led me to believe and I was concerned about.

The additional delay should be removed.

**Comment by Goldsteel** [comment]
This will be on hold for a little while. I’ll be able to continue in about a week.

**Comment by Kurfursten** [comment]
add the cooldown or mask of corgi will be insufferable

**Comment by SonixApache** [comment]
fix when

**Comment by Goldsteel** [comment]
In it's current state, there is a sanity check to prevent simple mobs from being spam petted for noises. They do not make noises when a player inhabits them, which I could fix, I guess. There didn't seem to be a delay on attack hand as I could spam patting noises.

**Comment by Goldsteel** [comment]
Fixed the thing. Now they say stuff when a player is in them, too.

**Comment by Coxswain-Navigator** [comment]
ok but will my cockroaches finally hiss intimidatingly? What about bees?

**Pull request merged**